### PR TITLE
Destination market's markups logic was added

### PR DESCRIPTION
**Pull request by DaniyarAskarov**
Issue: https://github.com/happy-travel/agent-app-project/issues/1290
Tested through insomnia and debug
Insomnia: Edo -> Admin -> Locations

Destination market markup defines by regionId if it equal to destinationScopeId. RegionId was determined by CountryCode which takes from mapper.

### CODE REVIEW

File: Api/Models/Accommodations/AccommodationAvailabilityResult.cs [hunk 1]
```diff
@@ -42,6 +43,6 @@ public record AccommodationAvailabilityResult
         public string SupplierAccommodationCode { get; init; }
         public string CountryHtId { get; init; }
         public string LocalityHtId { get; init; }
-
+        public int RegionId { get; init; }
```
Comment (AlexeyShok) [inline Api/Models/Accommodations/AccommodationAvailabilityResult.cs:46]: Why ReginId and not MarketId?

File: Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationPolicyProcessor.cs [hunk 2]
```diff
@@ -16,7 +16,8 @@ public static RoomContractSetAvailability Process(RoomContractSetAvailability av
                 accommodation: availability.Accommodation,
                 countryHtId: availability.CountryHtId,
                 localityHtId: availability.LocalityHtId,
-                evaluationToken: availability.EvaluationToken);
+                evaluationToken: availability.EvaluationToken,
+                regionId: availability.RegionId);
```
Comment (AlexeyShok) [inline Api/Services/Accommodations/Availability/Steps/BookingEvaluation/BookingEvaluationPolicyProcessor.cs:20]: Why ReginId and not MarketId?

**Review by kirillta — COMMENTED** [review]
Do we need a MarketId implementation so wide across availability models?

File: Api/Controllers/AdministratorControllers/LocationMarkupsController.cs [hunk 3]
```diff
@@ -66,9 +66,9 @@ public async Task<IActionResult> GetPolicies()
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
         [AdministratorPermissions(AdministratorPermissions.MarkupManagement)]
-        public async Task<IActionResult> ModifyPolicy(int policyId, [FromBody] MarkupPolicySettings policySettings)
+        public async Task<IActionResult> UpdatePolicy(int policyId, [FromBody] MarkupPolicySettings policySettings)
```
Comment (AlexeyShok) [inline Api/Controllers/AdministratorControllers/LocationMarkupsController.cs:69]: ModifyPolicy

File: Api/Controllers/AdministratorControllers/LocationMarkupsController.cs [hunk 4]
```diff
@@ -66,9 +66,9 @@ public async Task<IActionResult> GetPolicies()
...
         {
-            var (_, isFailure, error) = await _policyManager.ModifyLocationPolicy(policyId, policySettings);
+            var (_, isFailure, error) = await _policyManager.UpdateLocationPolicy(policyId, policySettings);
```
Comment (AlexeyShok) [inline Api/Controllers/AdministratorControllers/LocationMarkupsController.cs:71]: ModifyLocationPolicy

File: Api/Services/Accommodations/Availability/Mapping/AvailabilitySearchAreaService.cs [hunk 5]
```diff
@@ -51,20 +53,25 @@ static void FillSupplierCodes(in AccommodationMapping accommodationMapping, Dict
                         AccommodationHtId = accommodationMapping.HtId,
                         SupplierCode = supplierInfo.Value,
                         CountryHtId = location.CountryHtId,
-                        LocalityHtId = location.LocalityHtId 
+                        LocalityHtId = location.LocalityHtId,
+                        MarketId = _edoContext.Countries
```
Comment (AlexeyShok) [inline Api/Services/Accommodations/Availability/Mapping/AvailabilitySearchAreaService.cs:57]: Very suboptimal in terms of speed. It is better to load the entire list into the cache and then take from it

**Review by AlexeyShok — APPROVED** [review]

File: Api/Services/Markups/AdminMarkupPolicyManager.cs [hunk 6]
```diff
@@ -309,9 +397,16 @@ Task WriteGlobalLog(MarkupPolicyEventType eventType)
 
         private async Task<Result> Add(MarkupPolicyData policyData)
         {
-            var destinationScopeType = await GetDestinationScopeType(policyData.Settings.DestinationScopeId);
-            if (destinationScopeType.IsFailure)
-                return Result.Failure(destinationScopeType.Error);
+            var destinationScopeTypeValue = policyData.Settings.DestinationScopeType;
+
+            if (policyData.Settings.DestinationScopeType is null)
```
Comment (kirillta) [inline Api/Services/Markups/AdminMarkupPolicyManager.cs:402]:
```csharp
someType? destinationScopeTypeValue = null;

if (policyData.Settings.DestinationScopeType is null)
```
looks better

**Pull request merged**